core/templates/services/local-storage.service.ts

```diff
@@ -195,6 +199,125 @@ export class LocalStorageService {
     }
     return null;
   }
+
+  /**
+   * Retrieves the entity editor browser tabs info stored in the local storage
+   * for opened tabs of a particular entity type depending upon the
+   * browser tabs info constant.
+   * @param entityEditorBrowserTabsInfoConstant The key of the data stored on
+   * the local storage. It determines the type of the entity
+   * (topic, story, exploration, skill) for which we are retrieving data.
+   * @param entityId The id of the entity.
+   * @returns EntityEditorBrowserTabsInfo if the data is found on the local
+   * storage. Otherwise, returns null.
+   */
+  getEntityEditorBrowserTabsInfo(
+      entityEditorBrowserTabsInfoConstant: string, entityId: string
+  ): EntityEditorBrowserTabsInfo | null {
+    if (this.isStorageAvailable()) {
+      const allEntityEditorBrowserTabsInfoObjects:
+        EntityEditorBrowserTabsInfoObject[] = JSON.parse((
+        this.storage as Storage).getItem(
+          entityEditorBrowserTabsInfoConstant) || '[]');
+
+      const requiredEntityEditorBrowserTabsInfoObject = (
+        allEntityEditorBrowserTabsInfoObjects.find(
+          (entityEditorBrowserTabsInfoObject) => {
+            return entityEditorBrowserTabsInfoObject.id === entityId;
```

Comment:
I wonder a bit why you store this as a list rather than a dict? If you mapped ids to objects then would that enable a quicker lookup? (Or is ordering important for some reason?)

Reply:
Yes, it would improve. Changed it now.